Home_Task_13_1.py

```diff
@@ -0,0 +1,121 @@
+class MyCard:
+
+    def __init__(self, card_num, name, surname, expare_month, expare_year, cvv, pin, balance=0):
+        self.set_card_num(card_num)
+        self.set_person(name, surname)
+        self.set_expare_month(expare_month)
+        self.set_expare_year(expare_year)
+        self.set_cvv(cvv)
+        self.set_pin(pin)
+        self.balance = balance
+
+    def set_card_num(self, card_num):
+        if self.valid_number(card_num):
+            self.card_num = card_num
+        else:
+            raise ValueError("The card must be a 16-digit number.")
+
+    def get_card_num(self):
+        return self.card_num
+
+    @staticmethod
+    def valid_number(card_num):
+        return isinstance(card_num, int) and len(str(card_num)) == 16
+
+# __________person__________
+    def set_person(self, name, surname):
+        if self.valid_person(name) and self.valid_person(surname):
+            self.name = name
+            self.surname = surname
+        else:
+            raise ValueError("First name  and surname must contain only Latin characters and be up to 20 characters long.")
+
+    def get_name(self):
+        return self.name
+
+    def get_surname(self):
+        return self.surname
+
+    @staticmethod
+    def valid_person(text_name):
+        return text_name.isalpha() and len(text_name) <= 20
+
+# _________expare_month___________
+    def set_expare_month(self, expare_month):
+        if self.valid_month(expare_month):
+            self.expare_month = expare_month
+        else:
+            raise ValueError("expare_month must be a number between 1 and 12")
+
+    @staticmethod
+    def valid_month(expare_month):
+        return isinstance(expare_month, int) and 0 < expare_month < 13
+
+    def get_expare_month(self):
+        return self.expare_month
+
+# ________expare_year_________
+    def set_expare_year(self, expare_year):
+        if self.valid_year(expare_year):
+            self.expare_year = expare_year
+        else:
+            raise ValueError("expare_year must be a number between 2024 and 2029")
+
+    @staticmethod
+    def valid_year(expare_year):
+        return isinstance(expare_year, int) and 2023 < expare_year <= 2029
+
+    def get_expare_year(self):
+        return self.expare_year
+
+# ________сvv________
+    def set_cvv(self, cvv):
+        if self.valid_cvv(cvv):
+            self.cvv = cvv
+        else:
+            raise ValueError("CVV must have 3 digits")
+
+    @staticmethod
+    def valid_cvv(cvv):
+        return isinstance(cvv, int) and len(str(cvv)) == 3
+
+    def get_cvv(self):
+        return self.cvv
+
+# __________PIN___________
+    def set_pin(self, pin):
+        if self.valid_pin(pin):
+            self.pin = pin
```

Comment:
pay attention that pin and cvv , maybe balance too are protected fields. They should NOT be public